extensions/warp-ipfs/src/store/message/community_task.rs

```diff
@@ -1275,6 +1283,105 @@ impl CommunityTask {
 
         Ok(())
     }
+    async fn process_join_event(&mut self, msg: Message) -> Result<(), Error> {
+        let data = PayloadMessage::<Vec<u8>>::from_bytes(&msg.data)?;
+        let community_id = self.community_id;
+        let sender = data.sender().to_did()?;
+        let event = serde_json::from_slice::<CommunityJoinEvents>(data.message()).map_err(|e| {
+            tracing::warn!(community_id = %community_id, sender = %data.sender(), error = %e, "Failed to deserialize message");
+            e
+        })?;
+        match event {
+            CommunityJoinEvents::Join => {
+                let mut is_invited = false;
+                for (_, invite) in &self.document.invites {
+                    if let Some(expiry) = invite.expiry {
+                        if expiry < Utc::now() {
+                            continue;
+                        }
+                    }
+                    if let Some(target) = &invite.target_user {
+                        if &sender != target {
+                            continue;
+                        }
+                    }
+                    is_invited = true;
+                    break;
+                }
+                if !is_invited {
+                    self.send_single_community_event(
+                        &sender.clone(),
+                        ConversationEvents::JoinCommunity {
+                            community_id,
+                            community_document: None,
+                        },
+                    )
+                    .await?;
+                    return Ok(());
+                }
+
+                self.document.members.insert(sender.clone());
+
+                let mut invites_to_remove = vec![];
+                for (id, invite) in &self.document.invites {
+                    if let Some(target) = &invite.target_user {
+                        if &sender == target {
+                            invites_to_remove.push(id.clone());
+                        }
+                    }
+                }
+                for id in &invites_to_remove {
+                    self.document.invites.swap_remove(id);
+                }
+
+                self.set_document().await?;
+
+                self.send_single_community_event(
+                    &sender.clone(),
```

Comment:
Shouldnt need to borrow a clone. 

```suggestion
                    &sender,
```